conseil-common/src/main/scala/tech/cryptonomic/conseil/common/ethereum/EthereumPersistence.scala

```diff
@@ -61,24 +66,51 @@ class EthereumPersistence[F[_]: Concurrent] extends ConseilLogSupport {
     )
 
   /**
-    * Create [[DBIO]] seq with contracts.
+    * Create [[DBIO]] seq with contract account balances.
     *
-    * @param contracts JSON_RPC contract
+    * @param contractAccounts contract account eth_getBalance data
     */
-  def createContracts(contracts: List[Contract]) =
+  def createContractAccounts(contractAccounts: List[Account]): DBIOAction[Unit, NoStream, Effect.Write] =
     DBIO.seq(
-      Tables.Contracts ++= contracts.map(_.convertTo[Tables.ContractsRow])
+      Tables.Accounts ++= contractAccounts.map(_.convertTo[Tables.AccountsRow])
     )
 
   /**
-    * Create [[DBIO]] seq with tokens.
+    * Create [[DBIO]] seq with account balances.
     *
-    * @param logs JSON_RPC token
+    * @param accounts account eth_getBalance data
     */
-  def createTokens(tokens: List[Token]) =
+  def createAccountBalances(accounts: List[Account]): DBIOAction[Unit, NoStream, Effect.Write] =
     DBIO.seq(
-      Tables.Tokens ++= tokens.map(_.convertTo[Tables.TokensRow])
+      Tables.AccountsHistory ++= accounts.map(_.convertTo[Tables.AccountsHistoryRow])
+    )
+
+  /**
+    * Create [[DBIO]] seq with updated account balances.
+    *
+    * @param accounts account eth_getBalance data
+    */
+  def upsertAccounts(accounts: List[Account])(
+      implicit ec: ExecutionContext
+  ): DBIOAction[List[Unit], NoStream, Effect.Read with Effect.Write with Effect.Transactional] = {
+    import tech.cryptonomic.conseil.common.sql.CustomProfileExtension.api._
+    DBIO.sequence(
```

Comment:
Maybe I don't understand something, but why didn't you use something like this: `Tables.Accounts.insertOrUpdateAll(accounts.map(_.convertTo[Tables.AccountsRow]))`?

Reply:
Basically I want to update Account with the latest data `(blockHash, blockNumber, timestamp, balance)` without overwriting contract/token data `(bytecode, bytecode_hash, token_standard, name, symbol, decimals, total_supply)`

That's why I take the existing row, update a subset of fields and write the updated row to DB